src/redux/Dispatch/actions.js

```diff
@@ -300,11 +303,24 @@ export function assignTask(task, username) {
 
     const httpClient = getState().app.httpClient
 
-    dispatch(assignTaskRequest())
+    const linkedTasks = withLinkedTasks(task, selectAllTasks(getState()))
+
+    if (linkedTasks.length > 1) {
+      dispatch(bulkAssignmentTasksRequest())
 
-    return httpClient.put(`${task['@id']}/assign`, { username })
-      .then(res => dispatch(assignTaskSuccess(res)))
-      .catch(e => dispatch(assignTaskFailure(e)))
+      return httpClient.put('/api/tasks/assign', {
+        username,
+        tasks: linkedTasks.map(t => t['@id']),
+      })
+        .then((responses) => dispatch(bulkAssignmentTasksSuccess(responses)))
+        .catch(e => dispatch(bulkAssignmentTasksFailure(e)))
```

Comment:
I think it's helpful to log such case to speed up debugging:

```
.catch(e => {
  console.error(e)
  dispatch(bulkAssignmentTasksFailure(e))
})
```

Otherwise, I can not find any traces of what went wrong. And the action also does not show anything useful:

![Screenshot 2024-03-13 at 15 27 35](https://github.com/coopcycle/coopcycle-app/assets/70273239/9aa41289-3e3a-41fc-a7b0-afbd67aeb5cc)

Reply:
oh! I think we dont this this in the UI because we do optimistic rendering, so the crash doesnt affect the user I think

so we should write `action.payload[0].assignedTo` -> `action.payload['hydra:member'][0].assignedTo` so we have the tasks

it is ok? or should we replace ['hydra:member'] key by something like `items` server side ?

Reply:
I dont have the `ERROR` events in my reactotron timeline I guess it is due to the fact I am using Linux app :/
I just the dispatched BULK_ASSIGNMENT_FAILURE action

Reply:
Did you change anything on the backend? I expect that `action.payload[0].assignedTo` worked at some point. Good to know why it got broken

Reply:
Do you log exception to the console?

```
.catch(e => {
  console.error(e)
  dispatch(bulkAssignmentTasksFailure(e))
})
```

Otherwise, there will be no errors in reactotron

Reply:
yes I think I log but didnt manage to see, I will double check tomorrow

Reply:
I didnt change the backend, it guess it has been broken all this time ? not sure

easiest way would be to check both repo when the function was first introduced in the app. I will check tomorrow